pkg/simple/client/openpitrix/helmwrapper/helm_wrapper.go

```diff
@@ -470,142 +344,107 @@ func (c *helmWrapper) Install(chartName, chartData, values string) (err error) {
 	} else {
 		if err.Error() == StatusNotFoundFormat {
 			// continue to install
-			return c.install(chartName, chartData, values, false)
+			return c.writeAction(chartName, chartData, values, false)
 		}
 		return err
 	}
 }
 
-func (c *helmWrapper) install(chartName, chartData, values string, upgrade bool) (err error) {
-	if klog.V(2) {
-		start := time.Now()
-		defer func() {
-			klog.V(2).Infof("run command end, namespace: %s, name: %s elapsed: %v", c.Namespace, c.ReleaseName, time.Now().Sub(start))
-		}()
-	}
+func (c *helmWrapper) mockRelease() (*helmrelease.Release, error) {
+	return helmrelease.Mock(&helmrelease.MockReleaseOptions{Name: c.ReleaseName, Namespace: c.Namespace}), nil
+}
 
-	if err = c.ensureWorkspace(); err != nil {
-		return
-	}
-	defer c.cleanup()
+func (c *helmWrapper) helmUpgrade(chart *chart.Chart, values map[string]interface{}) (*helmrelease.Release, error) {
+	upgrade := action.NewUpgrade(c.helmConf)
+	upgrade.Namespace = c.Namespace
 
-	err = c.setupPostRenderEnvironment()
-	if err != nil {
-		return
+	if c.dryRun || klog.V(8) == true {
+		upgrade.DryRun = true
 	}
-
-	if err = c.createChart(chartName, chartData, values); err != nil {
-		return
+	if len(c.labels) > 0 || len(c.annotations) > 0 {
+		postRenderer := newPostRendererKustomize(c.labels, c.annotations)
+		upgrade.PostRenderer = postRenderer
 	}
-	klog.V(8).Infof("namespace: %s, name: %s, chart values: %s", c.Namespace, c.ReleaseName, values)
 
-	stdout := &bytes.Buffer{}
-	stderr := &bytes.Buffer{}
-
-	cmd := exec.Cmd{
-		Path:   c.cmdPath,
-		Dir:    c.Workspace(),
-		Stdout: stdout,
-		Stderr: stderr,
-	}
+	return upgrade.Run(c.ReleaseName, chart, values)
+}
 
-	cmd.Args = make([]string, 0, 10)
+func (c *helmWrapper) helmInstall(chart *chart.Chart, values map[string]interface{}) (*helmrelease.Release, error) {
+	install := action.NewInstall(c.helmConf)
+	install.ReleaseName = c.ReleaseName
+	install.Namespace = c.Namespace
 
-	// only for mock
-	if c.mock {
-		cmd.Path = os.Args[0]
-		cmd.Args = []string{os.Args[0], "-test.run=TestHelperProcess", "--"}
-		cmd.Env = []string{"GO_WANT_HELPER_PROCESS=1"}
+	if c.dryRun || klog.V(8) == true {
```

Comment:
If the user runs ks-apiserver with option `-v 9`, this condition will always be true.